### DISCUSSION TITLE
Output depending on input, instead numbers appear on the output window and stuff appears on the input window.

### DISCUSSION THREAD

**Discussion by totordev** [Textualize/textual#2388] [Q&A] [answered]
Hey, I'm sorry for once again reporting something, but I have gotten an issue that is very weird. I've linked a video showing the bug.

https://user-images.githubusercontent.com/94982718/234628550-f2f7a674-53b9-4c6f-b97e-f5068702d557.mp4

Here's the code:
`#!/bin/python3
import subprocess
from textual.app import App, ComposeResult
from textual.containers import Container
from textual.reactive import reactive
from textual.widgets import Button, Header, Footer, Label, OptionList, Input, _option_list
class YeastApp(App):

    BINDINGS = [('q', 'quit', 'Quit')]
    CSS_PATH = "yeast.css"

    global prompt
    global selected_option

    prompt = ""
    selected_option = ""

    def on_option_list_option_selected(self, event: OptionList.OptionSelected) -> None:
        global selected_option
        if event.option_index == 0:
            selected_option = ""
        elif event.option_index == 1:
            selected_option = "-a"
        elif event.option_index == 2:
            selected_option = "-R"
        else:
            selected_option = "-aR"

    def on_input_changed(self, event: Input.Changed) -> None:
        global prompt
        prompt = event.value
        command = f"ls {selected_option} | grep {prompt}"
        self.query_one("#output").update('%s' %subprocess.call(f"{command}", shell=True))
    def action_quit(self) -> None:
        quit()
    def compose(self) -> ComposeResult:
        yield OptionList(
            "Set basic",
            "Set hidden",
            "Set recursive",
            "Set hidden recursive",
            classes="options"

        )
        yield Container(
            Container(
                Input(placeholder="Insert string here..."),
                Label("\nYeast is a terminal app created by\nshebangcode in 2023 using the\nTextual framework in Python. It's an app\nto search files, and it's very useful.\nIts mascot is... Growy! Growy is\na bread dough that uses yeast to grow."),
                classes="botwin"
            ),
            Label(f"{prompt}", classes="botwin", id="output"),
            classes="bot"
        )
if __name__ == '__main__':
    app = YeastApp()
    app.run()`

**Comment by davep** [comment]
This isn't really a Textual issue, but... from what I'm seeing in the video the code is working as you've coded it. By the looks if things you're wanting to capture the output when running a subprocess? If so, you probably want to have a read over the [Python subprocess documentation](https://docs.python.org/3/library/subprocess.html) to see how to do that.

**Reply by totordev** [reply]
I wanted to run a shell command inside of Python, and then output the result of this command in the output window

**Reply by TomJGooding** [reply, marked as answer]
From the docs for `subprocess.call`:

> Run the command described by args. Wait for command to complete, then return the returncode attribute.

Here's a quick demo which might help explain this:

```python3
import subprocess

pattern = input("Insert string: ")
command = f"ls | grep {pattern}"

print("##### START OF SUBPROCESS COMMAND #####")
output = subprocess.call(command, shell=True)
print("##### END OF SUBPROCESS COMMAND #####\n")

print(f"Output value: {output}")
print(f"Type of output: {type(output)}")
```

You want to **check** the **output** from the executed command -  perhaps there's a another subprocess function that would help here...?

**Reply by totordev** [reply]
![image](https://user-images.githubusercontent.com/94982718/234816744-b9c4c772-d770-414e-9b70-27ce734a2ec0.png)
My code editor tells me that `command` is of type `int`, but it's supposed to be of type `str`. Is it possible to fix this?

**Reply by davep** [reply]
As @TomJGooding quotes above, [and can be seen in the Python documentation](https://docs.python.org/3/library/subprocess.html#subprocess.call), `subprocess.call` does not return a string; your editor is quite correct that it returns an `int`. There is nothing to fix here, this is how the call you're making to Python's library code works.

More generally; these aren't Textual questions or related to Textual. This is core Python. I'd strongly recommend having a read over the documentation linked to above and here.

**Reply by totordev** [reply]
Ok, I'll give it a read.
